### PR TITLE
[Impeller] minor optimization for glyph atlas sizing

### PR DESCRIPTION
**Pull request by dnfield**
If the current atlas size leaves fewer than half the remaining glyphs, only increase the size in one dimension.

This will not fare well in cases where there are very large glyphs remaining, but we plan to move very large glyphs out of this code anyway.

In the `textfield_fullscreen` benchmark in the framework, this reduces the atlas size by half (from 1024x1024 to 512x1024), saving 2mb of memory per frame (because the atlas has colored emojis in it).

### CODE REVIEW

File: impeller/typographer/typographer_unittests.cc [hunk 1]
```diff
@@ -93,5 +93,22 @@ TEST_P(TypographerTest, LazyAtlasTracksColor) {
...
+  SkFont sk_font;
+  auto blob = SkTextBlob::MakeFromString("AGH", sk_font);
+  ASSERT_TRUE(blob);
+  auto atlas = context->CreateGlyphAtlas(GlyphAtlas::Type::kAlphaBitmap,
+                                         TextFrameFromTextBlob(blob));
+  ASSERT_NE(atlas, nullptr);
+  ASSERT_NE(atlas->GetTexture(), nullptr);
+
+  // The 3 unique glyphs should not evenly fit into a square texture, so we
+  // should have a rectangular one.
```
Comment (dnfield) [inline impeller/typographer/typographer_unittests.cc:108]: I'm not super thrilled with this test, cause in theory three glyphs could fit evenly, but ... they don't.

**Comment by jonahwilliams** [comment]
Does reducing the atlas size improve performance, or is the CPU mostly busy actually painting the glyphs?

**Comment by dnfield** [comment]
I didn't measure the upload times, but it reduces memory usage.

**Review by jonahwilliams — APPROVED** [review]
Lgtm

**Pull request merged**